x-pack/plugin/core/src/main/java/org/elasticsearch/license/SelfGeneratedLicense.java

```diff
@@ -27,8 +27,8 @@
 
 class SelfGeneratedLicense {
 
-    public static License create(License.Builder specBuilder, DiscoveryNodes currentNodes) {
-        return create(specBuilder, LicenseUtils.compatibleLicenseVersion(currentNodes));
+    public static License create(License.Builder specBuilder, ClusterState currentState) {
```

Comment:
Sincd the license version doesn't depend on the cluster state, could it be removed from self generated create?